core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -429,7 +429,7 @@ private class InnerClosureFinder(output: Set[Class[_]]) extends ClassVisitor(ASM
           output += Class.forName(
               owner.replace('/', '.'),
               false,
-              Thread.currentThread.getContextClassLoader)
+              getClass.getClassLoader)
```

Comment:
I see where you're going with it, and it makes some sense. This is a fairly big change so want to think it through first. For example I wonder, by the logic of this change, when you'd ever want the context classloader?

Here also you're using the classloader that loaded Spark; I'm not sure it can see user-defined classes? haven't thought this one through.